app/controllers/bookings_controller.rb

```diff
@@ -52,14 +52,33 @@ def scan_ticket
 
   def request_refund
     @booking = Booking.find_by_uniq_id(params[:uniq_id])
-    flash[:notice] = if @booking.update_attributes(
-      refund_requested: true,
-      time_requested: Time.now
-    )
-                       "Request for a refund has been sent"
+    unless @booking.refund_requested
+      @booking.update_attributes(
+        refund_requested: true,
+        time_requested: Time.now,
+        reason: params[:reason]
+      )
+    end
+  end
+
+  def grant_refund
+    @booking = Booking.find_by_uniq_id(params[:uniq_id])
+    data = {
+      granted: true,
+      granted_by: current_user.id,
+      time_granted: Time.now
+    }
+    flash[:notice] = if !@booking.granted
```

Comment:
**Critical - Code Quality**
This could be refactored to:
```
flash[:notice] = if @booking.granted
                   "This request has already been granted"
                 else
                    method_to_handle_other_conditions
                 end
```